allennlp/data/dataset_readers/dataset_utils/span_utils.py

```diff
@@ -0,0 +1,131 @@
+from typing import List, Tuple, Callable, TypeVar
+
+from allennlp.data.dataset_readers.dataset_utils.ontonotes import TypedStringSpan
+
+T = TypeVar("T")
+def enumerate_spans(sentence: List[T],
+                    offset: int = 0,
+                    max_span_width: int = None,
+                    min_span_width: int = None,
+                    filter_function: Callable[[List[T]], bool] = None) -> List[Tuple[int, int]]:
+    """
+    Given a sentence, return all token spans within the sentence. Spans are `inclusive`.
+    Additionally, you can provide a maximum and minimum span width, which will be used
+    to exclude spans outside of this range.
+
+    Finally, you can provide a function mapping ``List[T] -> bool``, which will
+    be applied to every span to decide whether that span should be included. This
+    allows filtering by length, regex matches, pos tags or any Spacy ``Token``
+    attributes, for example.
+
+    Parameters
+    ----------
+    sentence : ``List[T]``, required.
+        The sentence to generate spans for. The type is generic, as this function
+        can be used with strings, or Spacy ``Tokens`` or other sequences.
+    offset : ``int``, optional (default = 0)
+        A numeric offset to add to all span start and end indices. This is helpful
+        if the sentence is part of a larger structure, such as a document, which
+        the indices need to respect.
+    max_span_width : ``int``, optional (default = None)
+        The maximum length of spans which should be included. Defaults to len(sentence).
+    min_span_width : ``int``, optional (default = None)
+        The minimum length of spans which should be included. Defaults to 1.
+    filter_function : ``Callable[[List[T]], bool]``, optional (default = None)
+        A function mapping sequences of the passed type T to a boolean value.
+        If ``True``, the span is included in the returned spans from the
+        sentence, otherwise it is excluded..
+    """
+    max_span_width = max_span_width or len(sentence)
+    min_span_width = min_span_width or 1
+    filter_function = filter_function or bool
```

Comment:
You can do `filter_function = filter_function or (lambda x: True)` to get around the syntax problem.